src/dao.ts

```diff
@@ -584,10 +585,15 @@ const makeDao = async function(entityType: EntityType, options: DaoOptionsInput
 
     // TODO Fetch nested references: a -> b -> c. After fetching all of a's references, collect the newly added items,
     // and fetch references for each of them (preferably together instead of sequentially).
-    fetchRelationships: async function(items: Entity[], pathsToExpand?: string[], {client = undefined}: {client?: Client} = {}) {
+    fetchRelationships: async function(
+      items: Entity[],
+      pathsToExpand?: string[],
+      {client = undefined, entityTypeAtPathPrefix = undefined, pathPrefix = undefined}: {client?: Client, entityTypeAtPathPrefix?: EntityType, pathPrefix?: string} = {}
+    ) {
       if (pathsToExpand && pathsToExpand.length == 0) {
         return
       }
+      if (entityTypeAtPathPrefix !== undefined) entityType = entityTypeAtPathPrefix
```

Comment:
This line seems dangerous. I'd suggest defining a local const instead of changing the entityType belonging to the DAO.